src/stores/messages/MessagesDataProviderStore.ts

```diff
@@ -295,10 +296,20 @@ export default class MessagesDataProviderStore implements MessagesDataStore {
 		}
 
 		if (messages.length !== 0) {
+			const previousMessages = messages.filter(
+				message =>
+					timestampToNumber(message.timestamp) < timestampToNumber(this.messages[0].timestamp),
+			);
+			if (previousMessages.length !== 0) this.messages = [...this.messages, ...previousMessages];
```

Comment:
what if this.messages.length is 1000 and previous messages should go after first message.
in the list would look like [firstMessages, 999 messages, previousMessages]